ring-jetty-adapter/test/ring/adapter/test/jetty.clj

```diff
@@ -115,6 +120,42 @@
       (is (thrown-with-msg? ConnectException #"Connection refused"
                             (http/get test-url)))))
 
+  (testing "HTTPS server that needs client certs"
+    (with-server client-cert-handler {:client-auth :need
+                                      :keystore "test/keystore.jks"
+                                      :key-password "password"
+                                      :port test-port
+                                      :ssl? true
+                                      :ssl-port test-ssl-port}
+      (is (thrown? Exception (http/get test-ssl-url {:insecure? true}))
```

Comment:
Is it possible to be a little more precise about the type of exception thrown?

Reply:
That's what I should've done, but I find a possible issue that there might be a race condition which results in a different exception is thrown each time, i.e.
- `javax.net.ssl.SSLHandshakeException: Received fatal alert: bad_certificate`
- `java.net.SocketException: Broken pipe (Write failed)`

So I used a general `Exception` to cover the tests cases in order to pass the CI. Do you have any insights for this?

Reply:
What about checking for `java.io.IOException`? Both exceptions inherit from that base.